### PR TITLE
Minor fixes and improvements of e2etest projects

### PR DESCRIPTION
**Pull request by dhei**
Fixes a few compiler warning in Xcode8. 

Makes local debugging of ZumoE2ETestApp.xcodeproject work out-of-box, not necessary to drag-and-drop MicrosoftAzureMobile.xcodeproject to ZumoE2ETestApp.

### CODE REVIEW

**Comment by dhei** [comment]
@shrishrirang

File: sdk/MicrosoftAzureMobile.xcodeproj/xcshareddata/xcschemes/MicrosoftAzureMobile.xcscheme [hunk 1]
```diff
@@ -1,6 +1,6 @@
 <?xml version="1.0" encoding="UTF-8"?>
 <Scheme
-   LastUpgradeVersion = "0710"
+   LastUpgradeVersion = "0800"
    version = "1.7">
```
Comment (shrishrirang) [inline sdk/MicrosoftAzureMobile.xcodeproj/xcshareddata/xcschemes/MicrosoftAzureMobile.xcscheme]: Did you have to edit this file manually?

File: e2etest/ZumoE2ETestApp.xcodeproj/project.pbxproj [hunk 2]
```diff
@@ -394,13 +406,25 @@
...
 				GCC_C_LANGUAGE_STANDARD = gnu99;
+				GCC_NO_COMMON_BLOCKS = YES;
+				GCC_WARN_64_TO_32_BIT_CONVERSION = YES;
 				GCC_WARN_ABOUT_RETURN_TYPE = YES;
+				GCC_WARN_UNDECLARED_SELECTOR = YES;
 				GCC_WARN_UNINITIALIZED_AUTOS = YES;
+				GCC_WARN_UNUSED_FUNCTION = YES;
 				GCC_WARN_UNUSED_VARIABLE = YES;
 				IPHONEOS_DEPLOYMENT_TARGET = 8.0;
 				OTHER_CFLAGS = "-DNS_BLOCK_ASSERTIONS=1";
```
Comment (shrishrirang) [inline e2etest/ZumoE2ETestApp.xcodeproj/project.pbxproj]: Did xcode modify the pbxproj file for you?

File: e2etest/ZumoE2ETestApp.xcodeproj/project.pbxproj [hunk 3]
```diff
@@ -41,6 +41,44 @@
...
+		9CA1216D1DAB1127000C2FCA /* PBXContainerItemProxy */ = {
+			isa = PBXContainerItemProxy;
+			containerPortal = 9CA1215C1DAB1127000C2FCA /* MicrosoftAzureMobile.xcodeproj */;
+			proxyType = 2;
+			remoteGlobalIDString = F3C002441AF7FF4A0028AB81;
+			remoteInfo = MicrosoftAzureMobile_MacTests;
+		};
+/* End PBXContainerItemProxy section */
+
 /* Begin PBXFileReference section */
```
Comment (shrishrirang) [inline e2etest/ZumoE2ETestApp.xcodeproj/project.pbxproj]: Don't think the default should be to include the SDK project in the e2e proejct. That is a development-only scenario, right?

File: e2etest/ZumoE2ETestApp/ZumoMainTableViewController.m [hunk 4]
```diff
@@ -73,7 +73,7 @@ - (UITableViewCell *)tableView:(UITableView *)tableView cellForRowAtIndexPath:(N
     }
 
     ZumoTestGroup *testGroup = [[self testGroups] objectAtIndex:[indexPath row]];
-    [[cell textLabel] setText:[NSString stringWithFormat:@"%ld. %@", [indexPath row] + 1, [testGroup name]]];
+    [[cell textLabel] setText:[NSString stringWithFormat:@"%ld. %@", (long)[indexPath row] + 1, [testGroup name]]];
```
Comment (shrishrirang) [inline e2etest/ZumoE2ETestApp/ZumoMainTableViewController.m]: Changing %ld to %d should fix the warning.

Comment (dhei) [inline e2etest/ZumoE2ETestApp/ZumoMainTableViewController.m]: `[indexPath row]` is NSInteger type which is either int or long, depending on the target platform. I am getting a warning on 64-bit platform when not casting it to long. According to [this](http://stackoverflow.com/a/26725692) stackoverflow question, casting NSInteger to long should be an acceptable solution.

File: sdk/MicrosoftAzureMobile.xcodeproj/xcshareddata/xcschemes/MicrosoftAzureMobile.xcscheme (hunk 1 above)
Comment (dhei) [inline sdk/MicrosoftAzureMobile.xcodeproj/xcshareddata/xcschemes/MicrosoftAzureMobile.xcscheme]: No, they are updated by Xcode automatically. xcscheme files are already checked in, so I don't see any problem updating them.

File: e2etest/ZumoE2ETestApp.xcodeproj/project.pbxproj (hunk 2 above)
Comment (dhei) [inline e2etest/ZumoE2ETestApp.xcodeproj/project.pbxproj]: Yes, I followed Xcode recommended build settings and let Xcode 8 did it for me. Did you spot anything problematic here?

**Comment by shrishrirang** [comment]
:shipit:

**Pull request merged**